cpp/modmesh/view/RMainWindow.cpp

```diff
@@ -30,17 +30,136 @@
 
 #include <modmesh/view/R3DWidget.hpp>
 
+#include <modmesh/view/RAction.hpp>
+#include <modmesh/view/RApplication.hpp>
+#include <Qt>
+#include <QMenuBar>
+#include <QMenu>
+#include <QAction>
+#include <QActionGroup>
+
 namespace modmesh
 {
 
+RMainWindow::RMainWindow()
+    : QMainWindow()
+{
+    // Do not call setUp() from the constructor.  Windows may crash with
+    // "exited with code -1073740791".  The reason is not yet clarified.
+}
+
 void RMainWindow::setUp()
 {
+    if (m_already_setup)
+    {
+        return;
+    }
+
+    this->setWindowIcon(QIcon(QString(":/icon.ico")));
+
+    this->setMenuBar(new QMenuBar(nullptr));
+    m_fileMenu = this->menuBar()->addMenu(QString("File"));
+    m_appMenu = this->menuBar()->addMenu(QString("App"));
+    m_cameraMenu = this->menuBar()->addMenu(QString("Camera"));
+    // NOTE: All menus need to be populated or Windows may crash with
```

Comment:
I troubleshooted this on the github actions CI and no way to use a debugger to debug it.